app/main.py

```
@@ -20,13 +22,30 @@ def sha256_hash_str(to_hash: str) -> str:
     return sha256(to_hash.encode("utf-8")).hexdigest()
 
 
-def brute_force_password() -> None:
-    pass
+def brute_force_password(start: int, end: int) -> None:
+    for i in range(start, end):
+        password = str(i).zfill(8)
+        if sha256_hash_str(password) in PASSWORDS_TO_BRUTE_FORCE:
+            print(password)
+
+
+def multi_brute_force():
+    tasks = []
+    count_cpu = multiprocessing.cpu_count() - 1
+    range_size = 100_000_000 // count_cpu
+
+    with ProcessPoolExecutor(count_cpu) as executor:
+        for i in range(count_cpu):
+            start = i * range_size
+            end = start + range_size
+            tasks.append(executor.submit(brute_force_password, start, end))
+
+    wait(tasks)
```

The `wait(tasks)` function is correctly used to wait for all futures to complete. However, ensure that you handle any exceptions that might be raised during the execution of tasks to avoid silent failures.